### PR TITLE
feat(binder): egg binder for drop statement

### PR DESCRIPTION
**Pull request by eliasyaoyc**
#707

### CODE REVIEW

File: src/binder_v2/drop.rs [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    pub(super) fn bind_drop(
+        &mut self,
+        object_type: ObjectType,
+        if_exists: bool,
+        names: Vec<ObjectName>,
+        cascade: bool,
+    ) -> Result {
+        self.push_context();
+        let ret = self.bind_drop_internal(object_type, if_exists, names, cascade);
+        self.pop_context();
```
Comment (wangrunji0408) [inline src/binder_v2/drop.rs:84]: Drop statement won't be nested, so there's no need to push another context.

File: src/binder_v2/drop.rs [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    #[error("invalid number: {0}")]
+    InvalidNum(#[from] std::num::ParseIntError),
+    #[error("invalid bool: {0}")]
+    InvalidBool(#[from] std::str::ParseBoolError),
+}
+
+impl FromStr for BindDrop {
+    type Err = ParseDropError;
+
+    fn from_str(s: &str) -> RawResult<Self, Self::Err> {
```
Comment (wangrunji0408) [inline src/binder_v2/drop.rs:54]: `FromStr` is used for parsing an egg expression from a LISP-style S-expression. It's hard to design a format for such a structure and maybe we will never construct one from the string. So I suggest just returning an error in this function like:
```rust
impl FromStr for BindDrop {
    type Err = ();
    fn from_str(s: &str) -> Result<Self, Self::Err> { Err(()) }
}
```

**Review by wangrunji0408 — COMMENTED** [review]
Rest LGTM. Thanks!

**Review by wangrunji0408 — APPROVED** [review]

**Comment by skyzh** [comment]
Indeed I'm wondering why we're commenting out the original drop and add a new BoundDrop between the expressions. Please hold this PR a little while and I will refactor tonight.

**Comment by wangrunji0408** [comment]
> Indeed I'm wondering why we're commenting out the original drop and add a new BoundDrop between the expressions. Please hold this PR a little while and I will refactor tonight.

I thought both ways are acceptable. Given that the drop statement will unlikely be rewritten by rules, representing it as a single `BoundDrop` value can avoid parsing it back from the `Expr` in the executor. 🤔

**Review by skyzh — APPROVED** [review]
LGTM. Some futures plans in my mind:

* Why insert / create / drop are part of `pub enum Expr`? 🤣 Seems a little bit weird but I'm thinking if it is possible to make it more clear.
* I believe we only need to support Drop(Id) in RisingLight. We can remove cascade / remove_if_exist later.

**Comment by wangrunji0408** [comment]
> Why insert / create / drop are part of pub enum Expr? 🤣 Seems a little bit weird but I'm thinking if it is possible to make it more clear.

Yeah, putting them all into the `Expr` is just for making a single AST representation in the planner, although these statements won't benefit much from the egg's features since there is no need to rewrite or analyze them.

**Pull request merged**